apps/minaintyg/src/hooks/useSessionWorker.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react'
+
+export const useSessionWorker = () => {
+  useEffect(() => {
+    const worker = new Worker(new URL('../workers/sessionWorker.js', import.meta.url), {
+      type: 'module',
+    })
+
+    worker.postMessage({ type: 'start', interval: 30000 })
+
+    worker.onmessage = (e) => {
+      if (e.data.type === 'logout') {
+        window.dispatchEvent(new Event('session-expired'))
```

Comment:
useSessionWorker could just take a callback function instead of triggering a event?

```typescript
export const useSessionWorker = (onLogout) => {
...
    worker.onmessage = (e) => {
      if (e.data.type === 'logout') {
          onLogout()
      }
    }
...
}
```